### DISCUSSION TITLE
Tag hierarchy

### DISCUSSION THREAD

**Discussion by BenjaminDeleze** [cyberbotics/webots-cloud#115] [Ideas]
As almost all the protos are added to webots.cloud, it is time to think about the tag hierarchy that we want to implement.

I looked at other sites of 3d assets and tried to adapt there structure to our assets, it would be something like that:

  - animals
  - appearance
  - architecture
    - building
    - street Furniture
    - traffic
    - road
  - device
    - lidar
    - distance sensor 
    - ...
  - object
    - house
      - bedroom
      - bathroom
      - kitchen
      - garden
      - apartement structure
      - light
      - living room
    - factory
      - container
      - tool
    - hospital
    - school
    - food and Drink
    - electronic
    - toy
 - plant
 - robot
    - arm
    - wheeled
    - flying
    - walking
  - vehicle
    - car
    - truck

I think we should allow some protos to have several tags like:
  - table: bedroom, living room, office

This is a first draft, do not hesitate to suggest modifications

**Comment by omichel** [comment]
I believe we wanted to make tags non-hierarchical, right?
I would make all keywords singular (e.g., animal, vehicle, plant).
Also, we should probably use lowercase for tag names.

**Reply by BenjaminDeleze** [reply]
> I would make all keywords singular (e.g., animal, vehicle, plant). Also, we should probably use lowercase for tag names.

Good point, I made the modifications

> I believe we wanted to make tags non-hierarchical, right?

No (it was my very very first idea some months ago), I think it is better to have a hierarchy for at least two reasons:
- The other website are working like that (https://www.turbosquid.com/3d-model/vehicle, https://free3d.com/3d-models/blender, https://www.cgtrader.com/3d-models)
- It is not really scalable and hard to find a particular tag if we have a huge list of all the tags at the same level.
